crates/trie/parallel/src/proof_task.rs

```diff
@@ -287,31 +396,47 @@ pub struct ProofTaskCtx {
     nodes_sorted: Arc<TrieUpdatesSorted>,
     /// The sorted in-memory overlay hashed state.
     state_sorted: Arc<HashedPostStateSorted>,
+    /// The collection of prefix sets for the computation. Since the prefix sets _always_
+    /// invalidate the in-memory nodes, not all keys from `state_sorted` might be present here,
+    /// if we have cached nodes for them.
+    prefix_sets: Arc<TriePrefixSetsMut>,
 }
 
 impl ProofTaskCtx {
     /// Creates a new [`ProofTaskCtx`] with the given sorted nodes and state.
     pub const fn new(
         nodes_sorted: Arc<TrieUpdatesSorted>,
         state_sorted: Arc<HashedPostStateSorted>,
+        prefix_sets: Arc<TriePrefixSetsMut>,
     ) -> Self {
-        Self { nodes_sorted, state_sorted }
+        Self { nodes_sorted, state_sorted, prefix_sets }
     }
 }
 
-/// A message used to send a storage proof request to the [`ProofTaskManager`].
+/// Message used to communicate with [`ProofTaskManager`].
 #[derive(Debug)]
 pub enum ProofTaskMessage<Tx> {
-    /// A storage proof request.
-    StorageProof((StorageProofInput, Sender<StorageProofResult>)),
+    /// A request to queue a proof task.
+    QueueTask(ProofTaskKind),
     /// A returned database transaction.
     Transaction(ProofTaskTx<Tx>),
     /// A request to terminate the proof task manager.
     Terminate,
 }
 
-/// A handle that wraps a single proof task sender that sends a terminate message on `Drop`.
+/// Proof task kind.
 #[derive(Debug)]
+pub enum ProofTaskKind {
+    /// A storage proof request.
+    StorageProof(StorageProofInput, Sender<StorageProofResult>),
+    /// A blinded account node request.
+    BlindedAccountNode(Nibbles, Sender<BlindedNodeResult>),
+    /// A blinded storage node request.
+    BlindedStorageNode(B256, Nibbles, Sender<BlindedNodeResult>),
+}
+
+/// A handle that wraps a single proof task sender that sends a terminate message on `Drop`.
+#[derive(Debug, Clone)]
```

Comment:
this is now clone, but still fires on drop. I don't think this is what we want here?